ScreenRecorderLibNative/Resizer.cpp

```diff
@@ -0,0 +1,257 @@
+    SampDesc.AddressU = D3D11_TEXTURE_ADDRESS_CLAMP;
+    SampDesc.AddressV = D3D11_TEXTURE_ADDRESS_CLAMP;
+    SampDesc.AddressW = D3D11_TEXTURE_ADDRESS_CLAMP;
+    SampDesc.ComparisonFunc = D3D11_COMPARISON_NEVER;
+    SampDesc.MinLOD = 0;
+    SampDesc.MaxLOD = D3D11_FLOAT32_MAX;
+    hr = m_Device->CreateSamplerState(&SampDesc, &m_SamplerLinear);
+    RETURN_ON_BAD_HR(hr);
+
+    // Create the blend state
+    D3D11_BLEND_DESC BlendStateDesc;
+    BlendStateDesc.AlphaToCoverageEnable = FALSE;
+    BlendStateDesc.IndependentBlendEnable = FALSE;
+    BlendStateDesc.RenderTarget[0].BlendEnable = TRUE;
+    BlendStateDesc.RenderTarget[0].SrcBlend = D3D11_BLEND_SRC_ALPHA;
+    BlendStateDesc.RenderTarget[0].DestBlend = D3D11_BLEND_INV_SRC_ALPHA;
+    BlendStateDesc.RenderTarget[0].BlendOp = D3D11_BLEND_OP_ADD;
+    BlendStateDesc.RenderTarget[0].SrcBlendAlpha = D3D11_BLEND_ONE;
+    BlendStateDesc.RenderTarget[0].DestBlendAlpha = D3D11_BLEND_ZERO;
+    BlendStateDesc.RenderTarget[0].BlendOpAlpha = D3D11_BLEND_OP_ADD;
+    BlendStateDesc.RenderTarget[0].RenderTargetWriteMask = D3D11_COLOR_WRITE_ENABLE_ALL;
+    hr = m_Device->CreateBlendState(&BlendStateDesc, &m_BlendState);
+    RETURN_ON_BAD_HR(hr);
+
+    // Initialize shaders
+    hr = InitShaders();
+    RETURN_ON_BAD_HR(hr);
+
+    return S_OK;
+}
+
+HRESULT Resizer::Resize(ID3D11Texture2D* orgTexture, ID3D11Texture2D** pResizedTexture, UINT targetWidth, UINT targetHeight, double viewPortRatio_width, double viewPortRatio_height)
+{
+    HRESULT hr;
+
+    // Create shader resource from texture of the original frame
+    D3D11_TEXTURE2D_DESC desktopDesc = {};
+    orgTexture->GetDesc(&desktopDesc);
+    D3D11_SHADER_RESOURCE_VIEW_DESC SDesc = {};
+    SDesc.Format = desktopDesc.Format;
+    SDesc.ViewDimension = D3D11_SRV_DIMENSION_TEXTURE2D;
+    SDesc.Texture2D.MostDetailedMip = desktopDesc.MipLevels - 1;
+    SDesc.Texture2D.MipLevels = desktopDesc.MipLevels;
+    ID3D11ShaderResourceView* srcSRV;
+    hr = m_Device->CreateShaderResourceView(orgTexture, &SDesc, &srcSRV);
+    if (FAILED(hr))
+    {
+        _com_error err(hr);
+        ERROR(L"Failed to create shader resource from original frame's texture: %ls", err.ErrorMessage());
+        return hr;
+    }
+
+    // Create target texture
+    CComPtr<ID3D11Texture2D> pResizedFrame = nullptr;
+    D3D11_TEXTURE2D_DESC targetDesc;
+    InitializeDesc(targetWidth, targetHeight, &targetDesc);
+    hr = m_Device->CreateTexture2D(&targetDesc, nullptr, &pResizedFrame);
+    RETURN_ON_BAD_HR(hr);
+    *pResizedTexture = pResizedFrame;
+    (*pResizedTexture)->AddRef();
+
+    // Make new render target view
+    ID3D11RenderTargetView* RTV;
+    hr = m_Device->CreateRenderTargetView(pResizedFrame, nullptr, &RTV);
+    RETURN_ON_BAD_HR(hr);
+
+    m_DeviceContext->OMSetRenderTargets(1, &RTV, nullptr);
+
+    // Set view port
+    SetViewPort(targetWidth, targetHeight, viewPortRatio_width, viewPortRatio_height);
```

Comment:
Should probably store old viewport with m_DeviceContext->RSGetViewports, and restore it to device context when done with drawing the resized texture.